### PR TITLE
[Diffusion] Zimage opt with qknorm and flashinfer rope

### PR DESCRIPTION
**Pull request by BBuf**
## SGLang generate main

sglang generate --model-path Tongyi-MAI/Z-Image-Turbo  --prompt "A cat is sitting on a chair" --tp-size 1 --height 1024 --width 1024

```shell
100%|██████████████████████████████████████████████████████| 9/9 [00:03<00:00,  2.26it/s]
[12-30 00:55:27] [DenoisingStage] average time per step: 0.4424 seconds
[12-30 00:55:27] [DenoisingStage] finished in 3.9857 seconds
```

sglang generate --model-path Tongyi-MAI/Z-Image-Turbo  --prompt "A cat is sitting on a chair" --tp-size 1 --height 2048 --width 2048

```shell
100%|██████████████████████████████████████████████████████| 9/9 [00:06<00:00,  1.34it/s]
[12-30 00:57:26] [DenoisingStage] average time per step: 0.7456 seconds
[12-30 00:57:26] [DenoisingStage] finished in 6.7150 seconds
```

## SGLang generate: qk norm + flashinfer rope

sglang generate --model-path Tongyi-MAI/Z-Image-Turbo  --prompt "A cat is sitting on a chair" --tp-size 1 --height 1024 --width 1024

```shell
100%|██████████████████████████████████████████████████████| 9/9 [00:02<00:00,  4.41it/s]
[12-30 02:05:34] [DenoisingStage] average time per step: 0.2269 seconds
[12-30 02:05:34] [DenoisingStage] finished in 2.0471 seconds
```

sglang generate --model-path Tongyi-MAI/Z-Image-Turbo  --prompt "A cat is sitting on a chair" --tp-size 1 --height 2048 --width 2048

```shell
100%|██████████████████████████████████████████████████████████████████| 9/9 [00:05<00:00,  1.54it/s]
[12-30 08:06:58] [DenoisingStage] average time per step: 0.6485 seconds
[12-30 08:06:58] [DenoisingStage] finished in 5.8410 seconds
```

图片

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) (`/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`) or contact authorized users to do so.
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/models/dits/zimage.py [hunk 1]
```diff
@@ -150,23 +154,40 @@ def forward(
...
+                k = self.norm_k(k)
 
         if freqs_cis is not None:
-            q = apply_rotary_emb(q, freqs_cis)
-            k = apply_rotary_emb(k, freqs_cis)
+            cos, sin = freqs_cis
+            if q.is_cuda and q.shape == k.shape:
+                cos_sin_cache = torch.cat(
+                    [
+                        cos.to(dtype=torch.float32).contiguous(),
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/models/dits/zimage.py:180]: nit: we are safe to remove `.to(dtype=torch.float32)` here
